nncf/experimental/tensorflow/patch_tf.py

```diff
@@ -0,0 +1,277 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import functools
+import inspect
+from typing import Optional
+from typing import List
+
+import tensorflow as tf
+from tensorflow.python.eager import context
+from tensorflow.python.framework import ops
+
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.experimental.tensorflow.nncf_context import get_nncf_context
+from nncf.experimental.tensorflow.scope import get_op_name
+from nncf.experimental.tensorflow.graph.argprovider import replace
+from nncf.experimental.tensorflow.nncf_operation import NNCFOperation
+from nncf.experimental.tensorflow.graph.transformations.commands import TFTargetPoint
+from nncf.experimental.tensorflow.graph.argprovider import TF_ARG_PROVIDERS
+
+
+class Hook:
+    """
+    Contains the NNCF operations and target point where
+    these operations should be applied.
+    """
+
+    def __init__(self,
+                 operations: List[NNCFOperation],
+                 target_point: TFTargetPoint):
+        """
+        Initializes the hook.
+
+        :param operations: List of the NNCF operations in the correct order.
+            The operation at index 0 is applied first, with index -1 last.
+        :param target_point: A target point. Contains information on where the
+            operations should be applied.
+        """
+        self._operations = operations
+        self._target_point = target_point
+
+        arg_provider_cls = TF_ARG_PROVIDERS.registry_dict.get(self._target_point.op_type_name)
+        if arg_provider_cls is None:
+            raise ValueError(f'Unexpected type of the TensorFlow operation: {self._target_point.op_type_name}. '
+                             'Register an `ArgProvider` instance for this type in the '
+                             '`TF_ARG_PROVIDERS` registry, please.')
```

Comment:
Ok, I see the runtime error, but not any develop-time checks (i.e. precommit-scope tests).